Classes/Plugin/Treeview.php

```diff
@@ -0,0 +1,304 @@
+                $collectionNames[] = Helper::getIndexNameFromUid(intval($collId), 'tx_dlf_collections', $this->conf['pages']);
+            }
+
+            // Last value is fake and used for distinction in $this->addCurrentCollection()
+            $params['filterquery'][]['query'] = 'collection_faceting:("'.implode('" OR "', $collIndexNames).'" OR "FakeValueForDistinction")';
+
+        }
+        $params['filterquery'][]['query'] = 'NOT type_sorting:"Periodical"';
+
+        $solr->params = $params;
+
+        if ($level == 0) {
+            foreach ($collectionNames as $collectionName) {
+                $newLevelArray[] = array("id" => "0##" . $collectionName, "text" => $collectionName, "children" => true, "a_attr" => array("href" => ""), "icon" => false);
+            }
+        } else {
+
+            $selectParams = array(
+                'rows' => 0
+            );
+
+            $querySelect = $solr->service->createSelect($selectParams);
+
+            $groupComponent = $querySelect->getGrouping();
+            $groupComponent->addField('year_sorting');
+
+            $field = $this->conf['field'];
+            $from = $this->conf['yearFrom'];
+            $till = $this->conf['yearTill'];
+            $gap = $this->conf['yearRange'];
+
+            if ($level == 1) {
+
+                for ($i = $from; $i <= $till; $i = $i + $gap) {
+                    $groupComponent->addQuery($field . ':[' . ($i - $gap) . ' TO ' . ($i - 1) . ']');
+                }
+            } else {
+                $groupComponent->addQuery($query);
+            }
+
+            $resultset = $solr->service->select($querySelect);
+            $groups = $resultset->getGrouping();
+
+            foreach ($groups as $groupKey => $group) {
+                $regex = $this->conf['regexOutput'];
+
+                preg_match($regex, $groupKey, $matches);
+
+                if ($level == 1) {
+
+                    unset($matches[0]);
+                    $text = implode(' ' . $this->conf['regexOutputDelimiter'] . ' ', $matches);
+
+                    if (!empty($text)) {
+
+                        $newLevelArray[] = array("id" => "1#" . $groupKey . "#".$groupKey,
+                            "text" => $text,
+                            "children" => true,
+                            "a_attr" => array("href" => ""),
+                            "icon" => false);
+                    }
+
+                } else {
+
+                    foreach ($group as $document) {
+
+                        $outputField = $this->conf['leafFieldOutput'];
+                        $text = $document->$outputField;
+
+                        $newLevelArray[] = array("id" => "2#" . $document->record_id . "#".$groupKey,
+                            "text" => $text,
+                            "children" => false,
+                            "a_attr" => array("href" => $document->purl),
```

Comment:
```suggestion
                            "a_attr" => ["href" => $document->purl],
```

`$document->purl` is not a required field! If you want to link to the document, please create a proper typolink to the PageView.